onnx/defs/tensor/defs.cc

```diff
@@ -1324,22 +1324,55 @@ ONNX_OPERATOR_SET_SCHEMA(
           auto& input_shape = getInputShape(ctx, 0);
           auto* output_shape = getOutputShape(ctx, 0);
           output_shape->clear_dim();
-          auto scales = ctx.getInputData(1);
+          auto scales = ctx.getInputData(1);         
           if (nullptr != scales) {
             // Infer output shape's dimension value if 'scales' is known.
-            if (scales->data_type() == TensorProto::FLOAT &&
-                scales->float_data_size() == input_shape.dim_size()) {
-              for (int i = 0; i < input_shape.dim_size(); ++i) {
-                float dim_value =
+            if (scales->data_type() == TensorProto::FLOAT) {
+              bool invalid_scale_shape = false;
+              std::vector<float> vec;
+              if (scales->has_raw_data()) {
+                const auto& data = ParseRawData<float>(scales);
+                vec.insert(vec.end(), data.begin(), data.end());
```

Comment:
I was following the same way we did in other ops. see https://github.com/onnx/onnx/blob/master/onnx/defs/tensor/defs.cc#L123 and L551.
Any specific reason to make assign() preferred?

Reply:
Because the vector here is brand new.
If it's not used afterward, you can just have something like
```
if (scales->has_raw_data()) { 
    const auto& data = ParseRawData<float>(scales); 
    vector<float> vec(data.begin(), data.end());
    ...
```
or if you want it later:
```
vector<float> vec;
if (scales->has_raw_data()) { 
    const auto& data = ParseRawData<float>(scales); 
    vec.assign(data.begin(), data.end());
    ...
```

This is a general suggestion, not something serious.
It's just a shorter way.